oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/LucenePropertyIndex.java

```diff
@@ -365,7 +365,7 @@ private boolean loadDocs() {
                             long f = PERF_LOGGER.start();
                             if (facetProvider == null) {
                                 facetProvider = new LuceneFacetProvider(
-                                        FacetHelper.getFacets(searcher, query, plan, indexNode.getDefinition().getSecureFacetConfiguration())
+                                        FacetHelper.getFacets(searcher, query, plan, indexNode.getDefinition().getSecureFacetConfiguration(), docs.totalHits)
```

Comment:
This `docs` object is first batch of results that fills up cursor. While `totalHits` might be accurate (it most likely is but I'd need to double check it), but I would prefer if we don't need to pass this on most likely because this requires an external agent to pass on an information that is actually implementation requirement and the external agent has a potential to mess up some internal assumption.

I understand with current code structure (switch deciding which type of facet counter to use quite early) it's easier to pass on this information. But, maybe, what we can do is to let `StatisticalSortedSetDocValuesFacetCounts#getTopChildren0` start doing its work and if it realizes that hit count is small enough then it can delegate to a new instance of `SortedSetDocValuesFacetCounts`. Wdyt?

Reply:
totalHits was accurate - that I checked - for instance - in case the query had 400 valid results but docs had only 50 top docs - then docs.size was equal to 50 but docs.totalHits was 400 ..I validated this with a couple of different use cases.

But yes - I agree with you , that it might be better if we don't pass total hits here (even I was a bit skeptical about this and wanted to get the approach reviewed). Let me check if we can work on a different approach as you suggested .